index.ts

```diff
@@ -0,0 +1,476 @@
+	next(): Item | true;
+}
+
+// moves.ts
+
+type MoveFlag = 1 | 0;
+
+interface MoveFlags {
+	authentic?: MoveFlag; // Ignores a target's substitute.
+	bite?: MoveFlag; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Strong Jaw.
+	bullet?: MoveFlag; // Has no effect on Pokemon with the Ability Bulletproof.
+	charge?: MoveFlag; // The user is unable to make a move between turns.
+	contact?: MoveFlag; // Makes contact.
+	dance?: MoveFlag; // When used by a Pokemon, other Pokemon with the Ability Dancer can attempt to execute the same move.
+	defrost?: MoveFlag; // Thaws the user if executed successfully while the user is frozen.
+	distance?: MoveFlag; // Can target a Pokemon positioned anywhere in a Triple Battle.
+	gravity?: MoveFlag; // Prevented from being executed or selected during Gravity's effect.
+	heal?: MoveFlag; // Prevented from being executed or selected during Heal Block's effect.
+	mirror?: MoveFlag; // Can be copied by Mirror Move.
+	mystery?: MoveFlag; // Unknown effect.
+	nonsky?: MoveFlag; // Prevented from being executed or selected in a Sky Battle.
+	powder?: MoveFlag; // Has no effect on Pokemon which are Grass-type, have the Ability Overcoat, or hold Safety Goggles.
+	protect?: MoveFlag; // Blocked by Detect, Protect, Spiky Shield, and if not a Status move, King's Shield.
+	pulse?: MoveFlag; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+	punch?: MoveFlag; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+	recharge?: MoveFlag; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+	reflectable?: MoveFlag; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+	snatch?: MoveFlag; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+	sound?: MoveFlag; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+	// single-target
+	'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+	// single-target, automatic
+	'self'|'randomNormal'|
+	// spread
+	'allAdjacent'|'allAdjacentFoes'|
+	// side and field
+	'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+	volatileStatus?: string;
+	boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+	chance?: number;
+	boosts?: Partial<BoostsTable>;
+	self?: SelfEffect;
+	status?: StatusName;
+	volatileStatus?: string;
+
+}
+interface Move extends Data {
+	kind: 'Move';
+
+	accuracy: number | true;
+	basePower: number;
+	flags: MoveFlags;
+	pp: number;
+	target: MoveTarget;
+	type: TypeName;
+
+	alwaysCrit?: boolean;
+	alwaysHit?: boolean;
+	basePowerModifier?: number;
+	boosts?: Partial<BoostsTable> | false
+	bypassesProtect?: boolean;
+	category: MoveCategory;
+	contestType?: string;
+	critModifier?: number;
+	critRatio?: number;
+	damage?: number | 'level' | false | null;
```

Comment:
i don't know what this type represents

Reply:
i believe `number` is for fixed damage (ie/ 40 for Dragon Rage), `level` is for things like Night Shade, and I dont really see false or null actually being used at all

Reply:
`false` and `null` are used by the game engine to make a move automatically fail or silent-fail. There's not all that much use-case in the data files, but you might want to set this in an `ActiveMove` to make it fail.